File: backend/src/models/Order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema(
  {
    orderId: { type: String, unique: true, required: true },
    customer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    farmer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    product: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Product',
      required: true,
    },
    quantity: { type: Number, required: true },
    unit: { type: String, required: true },
    pricePerUnit: { type: Number, required: true },
    totalAmount: { type: Number, required: true },
    orderDate: { type: String, required: true }, // YYYY-MM-DD
    deliveryDate: { type: String, required: true },
    deliverySlot: {
      type: String,
      enum: ['MORNING', 'EVENING', 'BOTH'],
      default: 'MORNING',
    },
    orderStatus: {
      type: String,
      enum: ['PENDING', 'CONFIRMED', 'PREPARING', 'READY', 'DELIVERED', 'SUPPLIED', 'CANCELLED', 'COMPLETED'],
      default: 'CONFIRMED',
    },
    paymentStatus: {
      type: String,
      enum: ['PAID', 'PARTIALLY_PAID', 'PENDING', 'OVERDUE'],
      default: 'PENDING',
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Order', orderSchema);

File: backend-dotnet/Models/Order.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DailyMarts.Api.Models;

[BsonIgnoreExtraElements]
public class Order
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonElement("orderId")]
    public string OrderId { get; set; } = string.Empty;

    [BsonElement("customer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Customer { get; set; } = string.Empty;

    [BsonElement("farmer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Farmer { get; set; } = string.Empty;

    [BsonElement("product")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Product { get; set; } = string.Empty;

    [BsonElement("quantity")]
    public double Quantity { get; set; }

    [BsonElement("unit")]
    public string Unit { get; set; } = "L";

    [BsonElement("pricePerUnit")]
    public decimal PricePerUnit { get; set; }

    [BsonElement("totalAmount")]
    public decimal TotalAmount { get; set; }

    [BsonElement("orderDate")]
    public string OrderDate { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-dd");

    [BsonElement("deliveryDate")]
    public string DeliveryDate { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-dd");

    [BsonElement("deliverySlot")]
    public string DeliverySlot { get; set; } = "MORNING"; // MORNING, EVENING, BOTH

    [BsonElement("orderStatus")]
    public string OrderStatus { get; set; } = "CONFIRMED"; // PENDING, CONFIRMED, PREPARING, READY, DELIVERED, SUPPLIED, CANCELLED, COMPLETED

    [BsonElement("paymentStatus")]
    public string PaymentStatus { get; set; } = "PENDING"; // PAID, PARTIALLY_PAID, PENDING, OVERDUE

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties for JSON population
    [BsonIgnore]
    public Product? ProductDetails { get; set; }

    [BsonIgnore]
    public User? FarmerDetails { get; set; }

    [BsonIgnore]
    public User? CustomerDetails { get; set; }
}
